src/v/archival/ntp_archiver_service.cc

```diff
@@ -2785,24 +2791,41 @@ ntp_archiver::find_reupload_candidate(manifest_scanner_t scanner) {
           segment_collector_mode::collect_non_compacted);
         auto candidate = co_await collector.make_upload_candidate(
           _conf->upload_io_priority, _conf->segment_upload_timeout);
-        if (candidate.candidate.exposed_name().empty()) {
-            vlog(_rtclog.warn, "Failed to make upload candidate");
+        if (const auto* error = std::get_if<candidate_creation_error>(
```

Comment:
Thoughts about `ss::visit` here?

Reply:
see https://github.com/redpanda-data/redpanda/pull/15089#discussion_r1403997928 - might need to think of some way around `ss::visit` + `upload_candidate_with_locks&&`

Reply:
changed to be same as https://github.com/redpanda-data/redpanda/pull/15089#discussion_r1404434486